lib/crypto/public_key_helper.c

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * X509 helper functions
+ *
+ * Copyright (c) 2024 Linaro Limited
+ * Author: Raymond Mao <raymond.mao@linaro.org>
```

Comment:
you have to preserve the original copyright from the functions you moved. So add that as well